asset-transfer-basic/chaincode-java/README.md

```diff
@@ -0,0 +1,164 @@
+
+## Basic asset transfer 
+
+This project demonstrates the use of the Java SDK, running a basic asset transfer contract using the "chaincode as a service" 
+pattern.
+
+
+## Setup 
+
+In this sample we will employ the [Kubernetes Test Network](../../test-network-k8s) to illustrate a scenario of 
+building, running, and debugging chaincode on a development workstation.  
+
+This project is also compatible with the legacy chaincode builder pipeline and the compose based test-network.
+For additional details, see the [End-to-end with the test-network](../../test-network/CHAINCODE_AS_A_SERVICE_TUTORIAL.md#end-to-end-with-the-the-test-network)
+documentation.
+
+## [Quickstart](../../test-network-k8s#quickstart)
+
+```shell
+export PATH=${PWD}/../../test-network-k8s:$PATH 
+
+network kind 
+
+network up 
+network channel create
+```
+
+```shell
+network chaincode deploy  ${PWD}
+
+network chaincode invoke  asset-transfer-basic '{"Args":["InitLedger"]}'
+network chaincode query   asset-transfer-basic '{"Args":["ReadAsset","asset1"]}' | jq 
+```
+
+## Detailed Guide 
+
+```shell
+network down
+network up
+network channel create
+```
+
+```shell
+# Build the chaincode docker image 
+docker build -t hyperledger/fabric-samples/asset-transfer-basic/chaincode-java .
+
+# Load the docker image directly to the KIND control plane.  
+# (Alternately, build/tag/push the image to a remote container registry, e.g. localhost:5000) 
+kind load docker-image hyperledger/fabric-samples/asset-transfer-basic/chaincode-java
+```
+
+```shell
+# Assemble the chaincode package archive 
+network chaincode package $PWD/ccpackage/ $PWD/build/asset-transfer.tgz
+
+# Determine the ID for the chaincode package 
+CORE_CHAINCODE_ID_NAME=$(network chaincode id $PWD/build/asset-transfer.tgz)
+
+# Launch the chaincode in k8s as Deployment + Service 
+network chaincode launch $PWD/build/asset-transfer.tgz
+
+# Complete the chaincode lifecycle 
+network chaincode install $PWD/build/asset-transfer.tgz 
+network chaincode approve asset-transfer-basic $CORE_CHAINCODE_ID_NAME
+network chaincode commit  asset-transfer-basic
+```
+
+```shell
+# execute the smart contract by name 
+network chaincode invoke  asset-transfer-basic '{"Args":["InitLedger"]}'
+network chaincode query   asset-transfer-basic '{"Args":["ReadAsset","asset1"]}'
+```
+
+```shell
+kubectl -n test-network logs -f deployment/org1peer1-cc-asset-transfer-basic
+```
+
+## Debugging 
+
+### Build
+
+```shell
+./gradlew shadowJar
+```
+or 
+```shell
+docker build -t fabric-samples/asset-transfer-basic/chaincode-java . 
+```
+
+
+### Package
+
+By instructing the peer to connect to chaincode at the Docker host alias `host.docker.internal`, pods running in 
```

Comment:
didn't use to work on Linux...

Reply:
oof.

https://stackoverflow.com/questions/48546124/what-is-linux-equivalent-of-host-docker-internal 

https://github.com/moby/libnetwork/pull/2348